types.go

```diff
@@ -445,45 +445,38 @@ func sprintName(s string) string {
 	var dst strings.Builder
 
 	for i := 0; i < len(s); {
-		if i+1 < len(s) && s[i] == '\\' && s[i+1] == '.' {
+		if s[i] == '.' {
 			if dst.Len() != 0 {
-				dst.WriteString(s[i : i+2])
+				dst.WriteByte('.')
 			}
-			i += 2
+			i++
 			continue
 		}
 
 		b, n := nextByte(s, i)
```

Comment:
`sprintTxtOctet` is very similar to this function so should probably changed as well.

Reply:
I don't see any changes that are necessary in `sprintTxtOctet`, because it always uses its string builder (avoiding inconsistency between internal code paths) and correctly handles the smaller set of characters that are special in strings as opposed to domain names (i.e., just ```\```, `"`, and those that are not printable ASCII). The choice to further preserve both unescaped `.` and escaped `\.` as they appear in the input is strange, but not harmful.